src/ctap2/mod.rs

```diff
@@ -1245,51 +1361,69 @@ pub(crate) fn credential_management(
 
 pub(crate) fn configure_authenticator(
     dev: &mut Device,
+    puat_result: Option<PinUvAuthResult>,
     cfg_subcommand: AuthConfigCommand,
     status: Sender<crate::StatusUpdate>,
     callback: StateCallback<crate::Result<crate::ManageResult>>,
     alive: &dyn Fn() -> bool,
-) {
+) -> bool {
     let mut authcfg = AuthenticatorConfig::new(cfg_subcommand);
     let mut skip_uv = false;
     let authinfo = match dev.get_authenticator_info() {
         Some(i) => i.clone(),
         None => {
             callback.call(Err(HIDError::DeviceNotInitialized.into()));
-            return;
+            return false;
         }
     };
 
     if authinfo.options.authnr_cfg != Some(true) {
         callback.call(Err(AuthenticatorError::HIDError(
             HIDError::UnsupportedCommand,
         )));
-        return;
+        return false;
     }
 
+    let mut skip_puap = false;
+    let mut pin_uv_auth_result = puat_result
+        .clone()
+        .unwrap_or(PinUvAuthResult::NoAuthRequired);
+    match puat_result {
+        Some(PinUvAuthResult::SuccessGetPinToken(t))
+        | Some(PinUvAuthResult::SuccessGetPinUvAuthTokenUsingUvWithPermissions(t))
+        | Some(PinUvAuthResult::SuccessGetPinUvAuthTokenUsingPinWithPermissions(t)) => {
+            skip_puap = true;
+            if authcfg.set_pin_uv_auth_param(Some(t)).is_err() {
+                return false;
```

Comment:
Probably need to send an error on the callback here?

Reply:
Yes, we should use `unwrap_result!()` here instead!